openlibrary/core/lists/model.py

```diff
@@ -97,12 +97,17 @@ def get_editions(self, limit=50, offset=0, _raw=False):
 
         When _raw=True, the edtion dicts are returned instead of edtion objects.
         """
-        # Anand - Dec 2014: Disabled it is wasn't working from long time.
+        edition_keys = set([
+                seed.key for seed in self.seeds
+                if seed and seed.type.key == '/type/edition'])
```

Comment:
Why do we have to check for `seed` first? Also, does this handle redirects correctly?